### DISCUSSION TITLE
Floating-point?

### DISCUSSION THREAD

**Discussion by jnoche** [mortbopet/Ripes#236] [Q&A] [answered]
Is the "F" standard extension for single-precision floating point currently supported in Ripes?

**Comment by mortbopet** [comment, marked as answer]
It is not - support for this would have to be implemented in:
1. The assembler - fairly trivial, due to how ISA extensions are defined in the assembler infrastructure.
2. Processor support - a bit less trivial. Would have to require some UI work (a new register file), processor model modifications (mainly around the register file to now indicate whether an instruction is writing into the I or F part of it, implementing new instructions in the ALU, control unit), and likewise updating the processor model views.

**Reply by jnoche** [reply]
Thank you for your response.
